packages/server/src/sdk/app/rows/search.ts

```diff
@@ -71,34 +72,43 @@ export async function search(
       options.sortOrder = options.sortOrder.toLowerCase() as SortOrder
     }
 
-    const table = await sdk.tables.getTable(options.tableId)
-    options = searchInputMapping(table, options)
+    let source: Table | ViewV2
+    let table: Table
+    if (options.viewId) {
+      source = await sdk.views.get(options.viewId)
+      table = await sdk.views.getTable(source)
+      options = searchInputMapping(table, options)
+    } else if (options.tableId) {
+      source = await sdk.tables.getTable(options.tableId)
+      table = source
+      options = searchInputMapping(table, options)
+    } else {
+      throw new Error(`Invalid source ID: ${options.viewId || options.tableId}`)
```

Comment:
Will this error message ever be correct? if both `options.viewId` and `options.tableId` are undefined then it'll just print `Invalid source ID: `

Reply:
Good catch, have made this error message better.